storm-client/src/jvm/org/apache/storm/metrics2/WindowedHistogram.java

```diff
@@ -1,38 +1,31 @@
-/*
- * Licensed to the Apache Software Foundation (ASF) under one
- * or more contributor license agreements.  See the NOTICE file
- * distributed with this work for additional information
- * regarding copyright ownership.  The ASF licenses this file
- * to you under the Apache License, Version 2.0 (the
- * "License"); you may not use this file except in compliance
- * with the License.  You may obtain a copy of the License at
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The ASF licenses this file to you under the Apache License, Version
+ * 2.0 (the "License"); you may not use this file except in compliance with the License.  You may obtain a copy of the License at
  *
  * http://www.apache.org/licenses/LICENSE-2.0
  *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
  */
 
-package org.apache.storm.metric.internal;
+package org.apache.storm.metrics2;
 
+import com.codahale.metrics.Histogram;
 import java.util.HashMap;
 import java.util.Map;
 import java.util.TimerTask;
-import org.apache.storm.metric.api.IMetric;
+import org.apache.storm.metric.internal.MetricStatTimer;
+import org.apache.storm.utils.Time;
 import org.apache.storm.utils.Utils;
 
 /**
- * Acts as a Latency Metric, but also keeps track of approximate latency for the last 10 mins, 3 hours, 1 day, and all time.
+ * Acts as a Histogram, but keeps track of approximate counts for the last 10 mins, 3 hours, 1 day, and all time.
  */
-public class LatencyStatAndMetric implements IMetric {
-    //The current lat and count buckets are protected by a different lock
```

Comment:
By removing this, are we saying use the same lock doesn't have much lock contention any more?

Reply:
I did not see the point of the existing complicated synchronization.  I can change it all back to the previous code if desired.

Reply:
I can see this separate currentLock can reduce lock contention. Otherwise `record` method might be blocked for too long by other methods 
https://github.com/apache/storm/blob/master/storm-client/src/jvm/org/apache/storm/metric/internal/LatencyStatAndMetric.java#L119-L123